substrafl/remote/register/manage_dependencies.py

```diff
@@ -0,0 +1,206 @@
+            extra_args: list = []
+            if lib_name == "substrafl":
+                extra_args = [
+                    "--find-links",
+                    dest_dir.parent / "substra",
+                    "--find-links",
+                    dest_dir.parent / "substratools",
+                ]
+            subprocess.check_output(
+                [
+                    sys.executable,
+                    "-m",
+                    "pip",
+                    "wheel",
+                    ".",
+                    "-w",
+                    LOCAL_WHEELS_FOLDER,
+                    "--no-deps",
+                ]
+                + extra_args,
+                cwd=str(lib_path),
+            )
+
+        shutil.copy(wheel_path, dest_dir / wheel_name)
+        wheel_names.append(wheel_name)
+
+    return wheel_names
+
+
+def get_pypi_dependencies_versions(lib_modules: List) -> List[str]:
+    """Retrieve the version of the PyPI libraries installed to generate the dependency list
+
+    Args:
+        lib_modules (list): list of modules to be installed.
+
+    Returns:
+        list(str): list of dependencies to install in the Docker container
+    """
+    return [f"{lib_module.__name__}=={lib_module.__version__}" for lib_module in lib_modules]
+
+
+def compile_requirements(dependency_list: List[str], *, operation_dir: Path, sub_dir: Path) -> None:
+    """Compile a list of requirements using pip-compile to generate a set of fully pinned third parties requirements
+
+    Writes down a `requirements.in` file with the list of explicit dependencies, then generates a `requirements.txt`
+    file using pip-compile. The `requirements.txt` file contains a set of fully pinned dependencies, including indirect
+    dependencies.
+
+    Args:
+        dependency_list: list of dependencies to install; acceptable formats are library names (eg "substrafl"),
+            any constraint expression accepted by pip("substrafl==0.36.0" or "substrafl < 1.0") or wheel names
+            ("substrafl-0.36.0-py3-none-any.whl")
+        operation_dir: path to the root dir
+        sub_dir: sub directory of the root dir where the `requirements.in` and `requirements.txt` files will be created
+
+    Raises:
+        IncompatibleDependenciesError: if pip-compile does not find a set of compatible dependencies
+
+    """
+    requirements_in = operation_dir / sub_dir / "requirements.in"
+
+    requirements = ""
+    for dependency in dependency_list:
+        if dependency.__str__().endswith(".whl"):
+            requirements += f"file:{dependency}\n"
+        else:
+            requirements += f"{dependency}\n"
+
+    requirements_in.write_text(requirements)
+    try:
+        subprocess.check_output(
+            [
+                sys.executable,
+                "-m",
+                "piptools",
+                "compile",
+                "--resolver=backtracking",
+                requirements_in,
+            ],
```

Comment:
What does this command do in case a invalid dependency ? like for instance a typo in "nupy" or smtgh like that ?
More generally, do we catch that kind of behaviour ?

Reply:
Might be good to add a test on this use case :)

Reply:
Okay, then might worth rename `IncompatibleDependenciesError` as `InvalidDependenciesError` to include both this use cases in the naming ?

Reply:
nupy exists actually 😅 https://pypi.org/project/nupy/
not sure if it's typo-spoofing or not 🤔